lib/associations/belongs-to-many.js

```diff
@@ -113,7 +113,7 @@ var BelongsToMany = function(source, target, options) {
     this.foreignKeyAttribute = {};
     this.foreignKey = this.options.foreignKey || Utils.camelizeIf(
       [
-        Utils.underscoredIf(this.source.options.name.singular, this.source.options.underscored),
+        Utils.underscoredIf(this.options.as || this.source.options.name.singular, this.source.options.underscored),
```

Comment:
Original behavior for `as` require us to use the `as` field entirely. If we change it to `N:M`, we have to change it to `1:N` as well?